src/gif.imageio/gifinput.cpp

```diff
@@ -361,6 +364,9 @@ GIFInput::read_subimage_data()
                 + (interlacing ? decode_line_number(wy, window_height) : wy);
         if (0 <= y && y < m_spec.height) {
             for (int wx = 0; wx < window_width; wx++) {
+                if (fscanline[wx] >= colormap_count) {
+                    return false;
```

Comment:
I think we can use an error message here to make it easier to understand why the read failed:

```suggestion
                    errorfmt("Possible corruption: Encoded value {:d} @ ({},{}) exceeds palette size {}\n",
                             fscanline[wx], wx, y, colormap_count);
                    return false;
```

Reply:
> I think we can use an error message here to make it easier to understand why the read failed:

Thanks for your reply. I have resubmitted it